### PR TITLE
ollama[major]: Integration package

### PR DESCRIPTION
**Pull request by bracesproul**
Draft mode until tool calling integration is added

### CODE REVIEW

**Comment by ronneldavis** [comment]
@bracesproul any relevant threads to follow for the tool calling integration?

**Comment by bracesproul** [comment]
> @bracesproul any relevant threads to follow for the tool calling integration?

Should come out sometime next week! I'd join the [Ollama discord](https://discord.com/invite/ollama), where they post releases/announcements in the `#announcements` channel

File: libs/langchain-ollama/langchain.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * @param {string} relativePath
+ * @returns {string}
+ */
+function abs(relativePath) {
+  return resolve(dirname(fileURLToPath(import.meta.url)), relativePath);
+}
+
+export const config = {
+  internals: [/node\:/, /@langchain\/core\//, 'ollama/browser'],
```
Comment (bracesproul) [inline libs/langchain-ollama/langchain.config.js:13]: cc @jacoblee93 I think I did this correctly? (`'ollama/browser'` was giving me rollup errors)

File: docs/core_docs/docs/integrations/chat/ollama_functions.mdx [hunk 2]
```diff
@@ -4,6 +4,10 @@ sidebar_label: Ollama Functions
 
 # Ollama Functions
 
+:::tip
```
Comment (jacoblee93) [inline docs/core_docs/docs/integrations/chat/ollama_functions.mdx:7]: Deprecate and make this a `warning` or whatever the red callout is

File: libs/langchain-ollama/langchain.config.js (hunk 1 above)
Comment (jacoblee93) [reply libs/langchain-ollama/langchain.config.js:13]: Yeah I think so

File: libs/langchain-ollama/package.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    "format": "prettier --config .prettierrc --write \"src\"",
+    "format:check": "prettier --config .prettierrc --check \"src\"",
+    "move-cjs-to-dist": "yarn lc-build --config ./langchain.config.js --move-cjs-dist",
+    "create-entrypoints": "yarn lc-build --config ./langchain.config.js --create-entrypoints",
+    "check-tree-shaking": "yarn lc-build --config ./langchain.config.js --tree-shaking"
+  },
+  "author": "LangChain",
+  "license": "MIT",
+  "dependencies": {
+    "@langchain/core": ">0.2.14 <0.3.0",
```
Comment (jacoblee93) [inline libs/langchain-ollama/package.json:38]: Let's just set to latest

File: libs/langchain-ollama/src/chat_models.ts [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+  tools?: (StructuredToolInterface | RunnableToolLike | ToolDefinition)[];
+}
+
+export interface PullModelOptions {
+  /**
+   * Whether or not to stream the download.
+   * @default true
+   */
+  stream?: boolean;
+  insecure?: boolean;
```
Comment (jacoblee93) [inline libs/langchain-ollama/src/chat_models.ts:48]: What's this?

File: libs/langchain-ollama/src/chat_models.ts [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+   * @default "http://127.0.0.1:11434"
+   */
+  baseUrl?: string;
+  /**
+   * Whether or not to check the model exists on the local machine before
+   * invoking it. If set to `true`, the model will be pulled if it does not
+   * exist.
+   * @default false
+   */
+  checkOrPullModel?: boolean;
```
Comment (jacoblee93) [inline libs/langchain-ollama/src/chat_models.ts:78]: Types aren't exported?

Comment (jacoblee93) [reply libs/langchain-ollama/src/chat_models.ts:78]: Would be nice to import this type

File: libs/langchain-ollama/src/chat_models.ts [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+    this.penalizeNewline = fields?.penalizeNewline;
+    this.streaming = fields?.streaming;
+    this.format = fields?.format;
+    this.keepAlive = fields?.keepAlive ?? this.keepAlive;
+    this.checkOrPullModel = fields?.checkOrPullModel ?? this.checkOrPullModel;
+  }
+
+  // Replace
+  _llmType() {
+    return "chat_ollama";
```
Comment (jacoblee93) [inline libs/langchain-ollama/src/chat_models.ts:261]: Just `ollama`

File: libs/langchain-ollama/src/chat_models.ts (hunk 4 above)
Comment (bracesproul) [reply libs/langchain-ollama/src/chat_models.ts:48]: `insecure ` is one of their fields. idk what it does but they offer it!

**Pull request merged**